### ISSUE TITLE
Search fields lack visual and programmatically associated labels [iOS]

### ISSUE THREAD

**Issue by willslone** [covid-alert-ny/covid-green-app#239]
The “search” text field found in the “Select your county” overlays relies on the “Search” placeholder text for a visual label and accessible name. Once a user has entered data in the field, the “search” label is gone and the field is simply announced as whatever data is in the field.

![Search text field focused with VoiceOver and announced as “Al”, which is the user inputted data](https://user-images.githubusercontent.com/4437084/92030068-82659880-ed34-11ea-9f70-f452067aa8eb.png)

I suggest adding a text label that remains visible at all times, then programmatically associate the label by including it in the accessibilityLabel attribute on the text field. Hide the visible text label from VoiceOver users via the accessibilityElementsHidden attribute, as not to announce the label twice to said users.

```javascript
<View accessibilityElementsHidden={true}>
  <Text>Search</Text>
</View>
<TextInput accessibilityLabel='Search' />
```

As an alternative to the visual text label, a “search” icon could be added to the beginning or end of the text field. The accessibilityLabel and accessibilityElementsHidden guidance from above would still apply.

**Comment by floridemai** [comment]
@willslone Can you check this in the next version and provide feedback. I am aware that search is not updated after entering a value... it probably should :)

**Comment by willslone** [comment]
@floridemai It is good for VoiceOver users now. It still needs some sort of persistent visible label (e.g., a search icon that doesn’t disappear when data is entered in the field). Lack of an always visible label is a problem for people with cognitive issues that affect short-term memory, as well as a more general usability issue.

**Comment by ckiss** [comment]
@willslone I added a permanently visible search icon to the input. Do you think is enough? should I add any text besides it?

![Screenshot_20200908_142741_gov ny health proximity](https://user-images.githubusercontent.com/1004239/92564835-7ae94680-f282-11ea-85d4-df60f5a1b470.jpg)

**Comment by willslone** [comment]
@ckiss The icon along with the accessible name of the field is sufficient. No extra text needed.

**Comment by ckiss** [comment]
fixed, avail in next build
